### ISSUE TITLE
Ability to list all compute with fully qualified DNS names (FQDN) in a compartment

### ISSUE THREAD

**Issue by sanketdjain** [terraform-providers/terraform-provider-oci#500]
### Terraform Version

Terraform v0.11.1
```
provider.null v1.0.0
provider.oci v2.0.5
provider.template v1.0.0
```

### OCI Provider Version

```
oci_v2.0.5 
2018/01/05 09:35:28 [INFO] terraform-provider-oci 2.0.5
```

### Description:
I want to list the exiting computes in the compartment along with the attributes such as:
- OCID
- DNS name (FQDN)
- Public IP
- Private IP
- Availability Domain
- Shape

Currently, I can obtain a list of instances using datasource oci_core_instances, but that only contains information about OCID, Availability domain and Shape from the above list. In order to list the DNS name and Public IP/Private IP, I have to list the VNICs, which in turn requires to list the VNIC Attachments.

Terraform HCL does not support recursive loops. So I don't see a way with the current architecture of OCI Terraform resources to list the Computes with the Public/Private IPs.

Can you please suggest if this is possible ?

**Comment by rcohenma** [comment]
The oci_core_instance resource already has that information

The VNIC data is not yet part of the oci_core_instances datasource so you cannot list compute instances with the Public/Private IPs

I will make a note of this request

**Comment by sanketdjain** [comment]
@rcohenma Any update on this request ? We really need this ability to generate a meaningful output from terraform, which can be consumed by other tools.

**Comment by rcohenma** [comment]
We are prioritizing this issue. 
We are working on finding the best solution to expose this information.

**Comment by alexng-canuck** [comment]
Hello @sanketdjain  ,

The latest version of Terraform v3.4.0 now has support for singular data sources that can get additional instance information such as private_ip, public_ip, etc...

To use it, you could specify something like this:

```
# Lists all instances in your compartment
data "oci_core_instances" "datasource" {
  compartment_id = "<instance_compartment_ocid>"
}

# Queries additional info for each instance individually
data "oci_core_instance" "singledatasource" {
  count = "${length(data.oci_core_instances.datasource.instances)}
  instance_id = "${data.oci_core_instances.datasource.instances.*.id[count.index]}"
}

# Output the private IPs for all instances
output "output" {
  value = "${data.oci_core_instance.singledatasource.*.private_ip}"
}
```

If you're using a pre-v3.0.0 provider, see this guide on how to upgrade to the latest:
https://www.terraform.io/docs/providers/oci/guides/version-3-upgrade.html

**Comment by alexng-canuck** [comment]
@sanketdjain , closing this issue since it's been addressed in v3.4.0 release. Let us know if you have any other questions.
